lib/puppet/configurer.rb

```diff
@@ -73,6 +74,10 @@ def retrieve_catalog(query_options)
     if (Puppet[:use_cached_catalog] && result = retrieve_catalog_from_cache(query_options))
       @cached_catalog_status = 'explicitly_requested'
 
+      Puppet.info _("Using cached catalog from environment '%{environment}'") % { environment: result.environment }
+    elsif (@pluginsync_failed && result = retrieve_catalog_from_cache(query_options))
+      @cached_catalog_status = 'on_pluginsync_failure'
```

Comment:
I noticed puppetserver#main CI is failing due to this change because the `cached_catalog_status_in_report` acceptance test is looking for `on_failure`, see https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppetserver/view/main/job/platform_puppetserver_integration-system_no-conditional_full-main/25/LAYOUT=ubuntu2004-64ma-64a,LDAP_TYPE=default,label=beaker/testReport/junit/(root)/ruby_puppet_acceptance_tests_reports/cached_catalog_status_in_report_rb/ I'm guessing it wasn't caught earlier because it's only started being a hard error in puppet 7?

In thinking about it some more, I'm not sure it's a good idea to create different failure statuses... For example, I think PE console uses at this field in the report to display different icons for success, failure, fell back to cache, etc.  (@jonathannewman can you confirm?) So I'm leaning towards having a generic `on_failure` and leaving it up to the user to look at the report (which will contain the error log messages).

Reply:
Yeah, the console is [explicitly looking for `on_failure`](https://github.com/puppetlabs/pe-console-ui/blob/c3873aa5a5c1fc7faa27c6ae547b470e3fe9c71e/src/puppetlabs/pe_console_middleware/helpers/status.clj#L45) so we shouldn't create more fine-grained failures.

Reply:
I filled https://github.com/puppetlabs/puppet/pull/8362 to revert the change